### PR TITLE
sig-network, Add networking tests to the sig network family

### PR DESCRIPTION
**Pull request by oshoval**
In order to have a method to run sig-network tests in an isolated environment,
mark the relevant tests with `sig-network`.

The dedicated job that has `sig-network` on its TARGET,
would run only the tests that are marked with `[sig-network]`.

CNAO is deployed on `sig-network` jobs.

See https://github.com/kubevirt/project-infra/pull/1007

Signed-off-by: Or Shoval <oshoval@redhat.com>

```release-note
None
```

### CODE REVIEW

File: tests/expose_test.go
Comment (phoracek) [inline tests/expose_test.go:51]: Use lower-case s.

File: tests/network/dual_stack_cluster.go
Comment (phoracek) [inline tests/network/dual_stack_cluster.go:12]: This and other modules are already tagged through SIGDescribe.

**Review by phoracek — CHANGES_REQUESTED** [review]

File: tests/expose_test.go
Comment (oshoval) [reply tests/expose_test.go]: Done (also s/N/n/g) as SIGDescribe has

File: tests/network/dual_stack_cluster.go
Comment (oshoval) [reply tests/network/dual_stack_cluster.go]: done

**Comment by oshoval** [comment]
Addressed comments, thanks
tested locally dry run

/cc @EdDev

**Comment by ormergi** [comment]
What you think about creating a script under automation/ for sig-network tests (instead of changing test.sh)?

File: automation/test.sh
Comment (EdDev) [inline automation/test.sh:43]: Why the `[]` are not included? Such a math may catch things lik `sig-networking`, no?

**Comment by oshoval** [comment]
> What you think about creating a script under automation/ for sig-network tests (instead of changing test.sh)?

Imo we don't need, as this is already the switch case that the other tests use, for sure not in this PR imo,
one that require, can do a refactor of automation/test.sh on a following PR.
one that need to run manually can simply use the good ole `KUBEVIRT_E2E_FOCUS="Sig-Network"`

**Review by EdDev — CHANGES_REQUESTED** [review]
- Please split it into two commits:
  - Add support for running only the sig-network lane.
  - Label the network tests.
- Please use the [standard](https://github.com/kubevirt/kubevirt/blob/e95b9bc92a45371f81fb2fa98b874de569ebbfb2/tests/network/framework.go#L26) labeling.
- I would prefer you do this on the PR that moves the tests, as you can easily use the labeler there(#5170).

File: automation/test.sh
Comment (oshoval) [reply automation/test.sh:43]: Its the `TARGET`, which comes from the prow job env var, its not the `[sig-network]` label if that what you means,

Comment (EdDev) [reply automation/test.sh:43]: Ahh right, this comment refers to line 291 then.

**Comment by EdDev** [comment]
> What you think about creating a script under automation/ for sig-network tests (instead of changing test.sh)?

Can you please explain what you mean?
I know that `test.sh` is the entry point of the CI jobs, so what exactly are you suggesting here?

File: automation/test.sh
Comment (oshoval) [reply automation/test.sh:43]: we never FOCUS `[X]`, but X and the test label either X or `[X]`
i don't think we should focus otherwise ?

Comment (EdDev) [reply automation/test.sh:43]: I do not understand you response. We have a specific label with square parenthesis and I gave an explicit example which may math as well (or not, you tell me).

Comment (oshoval) [reply automation/test.sh:43]: its a regex so it might catch sig-networking as well i believe,
but i meant that i don't see it as a problem, the labeling should be correct,
and we never used FOCUS that includes the `[]`

**Comment by oshoval** [comment]
> * Please split it into two commits:
>   
>   * Add support for running only the sig-network lane.
>   * Label the network tests.
> * Please use the [standard](https://github.com/kubevirt/kubevirt/blob/e95b9bc92a45371f81fb2fa98b874de569ebbfb2/tests/network/framework.go#L26) labeling.
> * I would prefer you do this on the PR that moves the tests, as you can easily use the labeler there(#5170).

About the two last points, in order to not create dependency between the PRs,
lets do it once one of them is merged ?
I don't see why its too acute to block this PR on other PR for this reason.
@phoracek @EdDev wdyt?

**Comment by ormergi** [comment]
> Can you please explain what you mean?
> I know that `test.sh` is the entry point of the CI jobs, so what exactly are you suggesting here? And why?

@EdDev I meant to create script for network-sig job instead of using automation/test.sh
In my opinion automation/test.shs this script is too big and going to grow even bigger 
(as there going to be other sig-* labels) I would try not to extend it more.

**Comment by EdDev** [comment]
> @EdDev I meant to create script for network-sig job instead of using automation/test.sh
> In my opinion automation/test.shs is too big and going to grow even bigger
> (as there going to be other sig-* labels) I would try not to extend it more.

So you are talking about splitting it into multiple shell scripts, one per job. Right?
That will be nice, but I think it requires splitting first the existing script and only then we can add an external script just for this new job.
Someone needs to do this eventually, I agree.

**Comment by EdDev** [comment]
> > * Please split it into two commits:
> >   
> >   * Add support for running only the sig-network lane.
> >   * Label the network tests.
> > * Please use the [standard](https://github.com/kubevirt/kubevirt/blob/e95b9bc92a45371f81fb2fa98b874de569ebbfb2/tests/network/framework.go#L26) labeling.
> > * I would prefer you do this on the PR that moves the tests, as you can easily use the labeler there(#5170).
> 
> About the two last points, in order to not create dependency between the PRs,
> given the current state of the branch, and the fact the there is work in progress (PR 5170 which exists but even not lgtmed approved yet and a lane fails there atm),
> lets do it once one of them is merged ?
> I don't see why its too acute to block this PR on other PR for this reason.
> @phoracek @EdDev wdyt?

This PR is not comlete and you have been feedbacked that it is less desired to use manual labeling.
You have been requested to use the existing labeling function which resides in the network subpackage, which implies that we need first to move the tests into that folder. I have not requested you to do the move, I already done it to save time.
I expect this to be done in a correct manner and the dependency seems a must to me.

You are welcome to provide alternatives.

**Comment by oshoval** [comment]
> This PR is not complete and you have been feedbacked that it is less desired to use manual labeling.
> You have been requested to use the existing labeling function which resides in the network subpackage, which implies that we need first to move the tests into that folder. I have not requested you to do the move, I already done it to save time.
> I expect this to be done in a correct manner and the dependency seems a must to me.
> 
> You are welcome to provide alternatives.

As i explained, imho creating dependency by PRs that are even not lgtm approve / total stable yet,
is not that crucial. It will happen eventually pretty soon.
Anyhow i would do that once you fix the flakes there please.

An alternative i am checking is using `network.SIGDescribe` is it fine by you if it works ?
Thanks

**Comment by oshoval** [comment]
Addressed the two comments in a separate commit that i will rearrange it looks good,
using label with `[]`
using network.SigDescribe

File: automation/test.sh
Comment (oshoval) [reply automation/test.sh:43]: Done (in a commit that i will rearrange in the end)

**Comment by EdDev** [comment]
> An alternative i am checking is using `network.SIGDescribe` is it fine by you if it works ?
> Edit - it works, please take a look on the pushed commit
> Thanks

Unfortunately we cannot import the network package into the tests, as this package includes tests and not libraries. Importing tests into other tests is not recommended.

**Comment by oshoval** [comment]
First push - address comments and rebase on #5170 
2nd push - rebase

**Comment by oshoval** [comment]
@EdDev 
can you suggest please about
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5167/pull-kubevirt-check-tests-for-flakes/1369251488844484608

i think its due to the rebase on top of #5170

**Comment by oshoval** [comment]
Rebase

File: automation/test.sh
Comment (EdDev) [inline automation/test.sh:291]: You use the escape char (`\`) once, why? I would expect it is either needed or it is not.

Comment (oshoval) [reply automation/test.sh:291]: it doesn't match without it, its a regex, that need it, the other `]` doesn't need to be escaped, 
i tested all the combinations

Comment (EdDev) [reply automation/test.sh:291]: I know why it is needed, I do not understand why you did not add it to the closing one for completeness.
It looks odd and raises questions.

**Review by EdDev — APPROVED** [review]
Thanks.

**Comment by oshoval** [comment]
Thanks Edy
/hold

until #5170 is merged
/cc @AlonaKaplan can you please take a look ?

File: automation/repeated_test.sh
Comment (jean-edouard) [reply automation/repeated_test.sh:96]: Nevermind, that change is from #5171, which is merged, not sure why it appears here, maybe this PR needs a rebase?

Comment (oshoval) [reply automation/repeated_test.sh:96]: Yes it needs, its on hold until both 5171 and 5170 will be merged (the first is merged as you said)
because it depends on those two

**Comment by oshoval** [comment]
Rebased
now depends only on #5170

**Comment by oshoval** [comment]
Need to rebase on 5170, on it

**Comment by oshoval** [comment]
Rebased on top of 5170

**Review by EdDev — APPROVED** [review]

**Comment by oshoval** [comment]
/lgtm cancel

It doesn't focus well (locally it worked)

**Comment by oshoval** [comment]
Changes:
Had to `\\` in order to catch the `[` of the filter as `hack/functests.sh` does
In order to be total aligned to it, we might want to consider double escaping also the `]`
as it does

**Comment by oshoval** [comment]
Something is still wrong, it worked in the last run (there was dryRun but now doesn't)

**Comment by oshoval** [comment]
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5167/pull-kubevirt-e2e-k8s-1.19-sig-network/1370010362480431104

Its fine now (maybe it was even fine before, i just misread it because some tests run in parallel),
FYI @EdDev that some tests run in parallel even on sig-network
(see please that it runs 72 and 21 tests in two batches)
Ran 21 of 750 Specs
Ran 72 of 750 Specs

**Comment by oshoval** [comment]
Rebase

**Comment by oshoval** [comment]
/test pull-kubevirt-e2e-k8s-1.19-sig-network

2/2

**Comment by EdDev** [comment]
> https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5167/pull-kubevirt-e2e-k8s-1.19-sig-network/1370010362480431104
> 
> Its fine now (maybe it was even fine before, i just misread it because some tests run in parallel),
> FYI @EdDev that some tests run in parallel even on sig-network
> (see please that it runs 72 and 21 tests in two batches)
> Ran 21 of 750 Specs
> Ran 72 of 750 Specs

You mean there are some tests that are not marked as serial?
I think we have not changed that by the move, but it is worth checking if this is problematic or not. At this stage I would not attempt to change it to serial if there is no good reason to do so. We may do this when we fully own the job.

**Comment by oshoval** [comment]
> > https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5167/pull-kubevirt-e2e-k8s-1.19-sig-network/1370010362480431104
> > Its fine now (maybe it was even fine before, i just misread it because some tests run in parallel),
> > FYI @EdDev that some tests run in parallel even on sig-network
> > (see please that it runs 72 and 21 tests in two batches)
> > Ran 21 of 750 Specs
> > Ran 72 of 750 Specs
> 
> You mean there are some tests that are not marked as serial?

Yep
> I think we have not changed that by the move, but it is worth checking if this is problematic or not. At this stage I would not attempt to change it to serial if there is no good reason to do so. We may do this when we fully own the job.

Indeed this move didn't touch the serialness of the tests, it was just FYI, since last week i think you said network doesn't have parallel tests, i won't touch it in this PR,
Thanks

**Comment by oshoval** [comment]
/test pull-kubevirt-e2e-k8s-1.19-sig-network

3/3 Passed

**Comment by oshoval** [comment]
/test pull-kubevirt-e2e-k8s-1.19-sig-network

4/4 Passed

**Comment by oshoval** [comment]
@EdDev 
How many runs would be enough ? so far 4/4 passed and the 5 is still on fire

**Comment by EdDev** [comment]
> @EdDev
> How many runs would be enough ? so far 4/4 passed and the 5 is still on fire

10 runs will be nice.

**Comment by oshoval** [comment]
6/6 Passed IIUC (not including the one that runs now)
correct me please if i am wrong, anyhow its no biggie

cloned here to run it parallel
https://github.com/kubevirt/kubevirt/pull/5219

**Comment by oshoval** [comment]
7/7 

/test pull-kubevirt-e2e-k8s-1.19-sig-network

**Comment by oshoval** [comment]
9/9 (one on https://github.com/kubevirt/kubevirt/pull/5219)

/test pull-kubevirt-e2e-k8s-1.19-sig-network

**Comment by oshoval** [comment]
10/10 Passed (pull-kubevirt-e2e-k8s-1.19-sig-network)

**Comment by oshoval** [comment]
Rebase

**Comment by oshoval** [comment]
/hold cancel

5170 was merged

**Review by EdDev — APPROVED** [review]

**Pull request merged**
